### DISCUSSION TITLE
disable_screensaver on a timer/schedule?

### DISCUSSION THREAD

**Discussion by semiligneous** [damongolding/immich-kiosk#74] [Ideas]
I wonder if anyone else would think it was useful to have the `disable_screensaver` only active at certain times of the day?

In the day time, I want the screen to show the kiosk slideshow, but at night, I want the computer's screensaver to kick in and turn off the display and lock eventually, i.e. `disable_screensaver=false`.

I could achieve this now by running kiosk with a URL with `disable_screensaver=true` in the daytime from 7am, and loading a different URL with `disable_screensaver=false` from 10pm onwards (there's probably a browser plugin to do this automatically...). But a schedule option like `disable_screensaver_schedule_start=0700` and `disable_screensaver_schedule_end=2200` would maybe be useful?

**Comment by semiligneous** [comment]
to be clear, I don't expect kiosk to wake up the computer from sleep and suddenly start playing the photo slideshow at 7am. I appreciate that is not possible! (Would be awesome if it could of course!).

It would require me, the user, to unlock the computer/cancel the screensaver, but nothing else would be needed - it would just start in screensaver_disabled mode.

**Comment by damongolding** [comment]
If there is enough interest I can implement that. 

Might be worth testing if requesting the wake lock would stop a screensaver if it was already playing. As you said, I couldn't unlock the device though.

**Comment by radh21301** [comment]
I think it would also be good enough if the screen can just turn black. Instead of turning off altogether (although turning off the screen has some power savings). As long as the bright light does not keep me from sleeping -> I use kiosk on a tablet in my bedroom, so I have to lock the tablet manually every night.

But I would actually prefer if I dont have to unlock or interact with the device everyday to turn it on in the morning. I prefer things to turn on/off automatically.

**Comment by damongolding** [comment]
If anyone is interested in being a beta tester they can use the docker image `damongolding/immich-kiosk-development:0.11.0` to test out the `sleep_start=TIME` and `sleep_start=TIME` features. If you do so and have any feedback you can either leave it here or hit me up on the Immich Discord server. There are some docs on it here [https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#sleep-mode](https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#sleep-mode)

**Reply by soana** [reply]
The sleep mode works fine for me, the image (date and time) is rather dim but it could be my monitor.

**Reply by radh21301** [reply]
It works great for me!

I have a really dark room at night, so the brightness is perfect for me.

**Reply by damongolding** [reply]
> The sleep mode works fine for me, the image (date and time) is rather dim but it could be my monitor.

Yeah it's a tough one. It's perfect on one of my monitors and too dim on the other.
